src/osx/cocoa/dataview.mm

```diff
@@ -163,6 +163,46 @@ -(id) copyWithZone:(NSZone*)zone
 }
 @end
 
+// ----------------------------------------------------------------------------
+// wxDVCNSHeaderView: exists only to override rightMouseDown:
+// ----------------------------------------------------------------------------
+
+@interface wxDVCNSHeaderView : NSTableHeaderView
+{
+    wxDataViewCtrl* dvc;
+}
+
+    -(id) initWithDVC:(wxDataViewCtrl*)ctrl;
+    -(void) rightMouseDown:(NSEvent *)theEvent;
+@end
+
+@implementation wxDVCNSHeaderView
+
+-(id) initWithDVC:(wxDataViewCtrl*)ctrl
+{
```

Comment:
I'm surprised you don't have to model along a initWithFrame initialization, since it is a NSView I would have thought the frame should be given already when initializing but apparently enough layout calls are triggered before things are shown ... and nothing vital seems to be in the NSTableHeaderView initWithFrame ...

Reply:
Thanks for looking at this!

I am not sure what I could/should pass to `initWithFrame:`, so I'm going to leave it like this for now, as not calling doesn't seem to result in any problems in practice...

Reply:
most of the NSViews get initialized with `InitWithFrame:` so this issue might arise elsewhere, that's why I thought I'd mention it. Here we'd need the headers frame, I've seen places that just calculate the height to be some constant (17). So if the need arises I'd do a `initWithFrame: DVC:` and call `[super initWithFrame:]`

Reply:
My problem is that I need to create this header view and I don't know which frame to pass to it when doing it. Currently I conveniently avoid the problem and I'd rather keep it like this until we find something actually requiring doing it.

Reply:
in this case, absolutely agreed, if it works that's fine